File: 1/solution.py
def calculateDepth(lines): 
	current = int(lines[0])
	depth = 0
	for line in lines: 
		if(int(line) > current):
			depth += 1
		current = int(line)
	return depth

def groupCalculate(lines): 
	depth = 0
	currentSum = 0
	previousSum = int(lines[0]) + int(lines[1]) + int(line[2])
	counter = 0
	for i in range(len(lines)):
		if counter + 3 < len(lines) :
			for it in range(3): 
				currentSum += int(lines[i])
		if currentSum > previousSum: 
			depth += 1
			previousSum = currentSum	
	return depth


file_name = "input.txt"
file_hook = open(file_name, 'r')
lines = file_hook.readlines()

result = calculateDepth(lines)
groupResult = groupCalculate(lines)
print("Depth: ", result)
print("Group depth ", groupResult)

File: 1/solution.cpp
#include <iostream>
#include <fstream>
#include <vector>


std::vector<int> getInput(std::string& filename) {

	std::ifstream stream(filename); 
	std::vector<int> input; 
	std::string line; 
	while(std::getline(stream, line))
	{
		input.push_back(std::stoi(line));
	}
	return input; 
}

int calculateDepth(std::vector<int>& input) {

	auto depth = 0;
	auto currentSum = 0;
	auto previousSum = 0; 
	for(int i = 0; i < input.size(); ++i)
	{
		if( (i + 2) < input.size())
		{
			currentSum = 0;
			for(int j = i; j < (i + 3); ++j)
			{

				currentSum += input[j]; 
			}

			if(previousSum < currentSum)
			{
				depth++; 
			}
			previousSum = currentSum;
		}
	}
	return depth - 1; 
}

auto main() -> int {

	std::string filename = "input.txt"; 
	auto input = getInput(filename); 
	auto result = calculateDepth(input);
	//auto result = 0; 
	std::cout<<"Result: "<< result << std::endl; 
	return 0;
}